clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -446,6 +448,9 @@ public void removeSensor(String name) {
                             removeMetric(metric.metricName());
                         log.debug("Removed sensor with name {}", name);
                         childSensors = childrenSensors.remove(sensor);
+                        for (final Sensor parent : sensor.parents()) {
```

Comment:
Hmm.. On sensor creation, we have a check if `parents` is null. Should we have that check here as well or is it unneeded in the creation logic?

Reply:
Not sure I follow: since we check and avoid nulls at the creation time already, `parents()` call is hence always safe to return not-null, right?